tutorials/ndvi_hotspots.ipynb

```diff
@@ -0,0 +1,1063 @@
+       "| 0.00,-100.00,-907900.00|\n",
+       "| 0.00, 0.00, 1.00|</dd><dt><span>resolution :</span></dt><dd>100</dd></dl></div></li></ul></div></div>"
+      ],
+      "text/plain": [
+       "<xarray.DataArray 'stackstac-335b61678a05cde2cc55dd18e3e138ef' (time: 5, band: 4, y: 1127, x: 1109)>\n",
+       "dask.array<where, shape=(5, 4, 1127, 1109), dtype=float64, chunksize=(1, 1, 1024, 1024), chunktype=numpy.ndarray>\n",
+       "Coordinates: (12/46)\n",
+       "  * time                                     (time) datetime64[ns] 2016-06-27...\n",
+       "    id                                       (time) <U65 'S2A_MSIL2A_20160627...\n",
+       "  * band                                     (band) <U5 'blue' 'green' ... 'nir'\n",
+       "  * x                                        (x) float64 1.362e+06 ... 1.472e+06\n",
+       "  * y                                        (y) float64 -9.079e+05 ... -1.02...\n",
+       "    s2:reflectance_conversion_factor         (time) float64 0.9679 ... 0.9695\n",
+       "    ...                                       ...\n",
+       "    title                                    (band) <U14 'Band 2 - Blue' ... ...\n",
+       "    proj:shape                               object {10980}\n",
+       "    common_name                              (band) <U5 'blue' 'green' ... 'nir'\n",
+       "    center_wavelength                        (band) float64 0.49 0.56 ... 0.842\n",
+       "    full_width_half_max                      (band) float64 0.098 ... 0.145\n",
+       "    epsg                                     int64 32619\n",
+       "Attributes:\n",
+       "    spec:        RasterSpec(epsg=32619, bounds=(1361700, -1020600, 1472600, -...\n",
+       "    crs:         epsg:32619\n",
+       "    transform:   | 100.00, 0.00, 1361700.00|\\n| 0.00,-100.00,-907900.00|\\n| 0...\n",
+       "    resolution:  100"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data = (\n",
+    "    stackstac.stack(\n",
+    "        items,\n",
+    "        epsg=32619,\n",
+    "        resolution=100,\n",
+    "        assets=['B02', 'B03', 'B04', 'B08']  # blue, green, red, nir\n",
+    "    )\n",
+    "    .where(lambda x: x > 0, other=np.nan)  # sentinel-2 uses 0 as nodata\n",
+    "    .assign_coords(band = lambda x: x.common_name.rename(\"band\"))  # use common names\n",
+    ")\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = data.persist()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "data = data.compute()"
```

Comment:
On the call we discussed whether or not to use Dask in this example. Right now we're just using it to parallelize the I/O. For reference, this took 2m30s to compute on a single-node with no Dask cluster. It took ~50s with a cluster (I'm a bit surprised it's that much slower; I would have expected the single client node to be I/O bound in either case, either straight from Blob Storage or from the various workers).

I'd recommend either using Dask to parallelize the computation too (so using `.persist` instead of `.compute`), or just skipping Dask entirely. Slight preference to use Dask, since 2m30s is a bit long to wait for results with nothing to look at.

Oh, but now I recall that performance issue with hotspots on Dask arrays being slow. Were you able to reproduce that? 

Either way, maybe increase the chunksize in `stackstac.stack` a bit: right now we end up with 4 chunks per timestep / band. It'd be a bit better to get that down to one:

![image](https://user-images.githubusercontent.com/1312546/119826637-d7969b00-bebd-11eb-89b5-d8028c8e78e5.png)